src/sync/installation.ts

```diff
@@ -114,12 +116,16 @@ export const updateJobStatus = async (
 
 	const updateRepoSyncFields: { [x: string]: string | Date} = { [getStatusKey(task)]: status };
 
-	if (isComplete && task === "commit" && data.commitsFromDate) {
+	//Set the complete date on success for tasks.
+	//Skip branches as it sync all history
+	if (isComplete && allTasksExceptBranch.includes(task) && data.commitsFromDate) {
 		const repoSync = await RepoSyncState.findByRepoId(subscription, repositoryId);
-		const commitsFromDate =  new Date(data.commitsFromDate);
-		// Set commitsFromDate in RepoSyncState only if its the later date
-		if (repoSync &&	(!repoSync.commitFrom || repoSync.commitFrom.getTime() > commitsFromDate.getTime())) {
-			updateRepoSyncFields["commitFrom"] = commitsFromDate;
+		if (repoSync) {
+			const newFromDate =  new Date(data.commitsFromDate);
```

Comment:
Should we rename the message payload field `commitsFromDate` to `fromDate`?

Reply:
I think that make sense. But I don't want to do it in this PR, and to transition it smoothly on the fly, ( becarefull of the existing msg in the queue), it will take quite some efforts.

I'd say let's wait a see whether future UI will allow user pick different dates for different tasks?